File: internal/tui/screens/help/help.go
package help

import (
	"strings"

	"github.com/charmbracelet/lipgloss"

	"scandoc/internal/tui/state"
	"scandoc/internal/tui/styles"
)

func Render(st *state.AppState) string {
	var b strings.Builder
	b.WriteString(styles.HeaderStyle.Render(" scanDOC Help & Keyboard Guide ") + "\n\n")

	b.WriteString(lipgloss.NewStyle().Bold(true).Foreground(styles.AccentColor).Render("Keyboard Navigation Cheat Sheet:") + "\n\n")

	shortcuts := [][2]string{
		{"1 .. 9", "Quick navigate to menu options"},
		{"w / s or ↑ / ↓", "Navigate up and down through list options"},
		{"Enter", "Select focused menu item or confirm action"},
		{"Space", "Toggle setting / multi-select file"},
		{"Ctrl+O / o", "Open Document File Browser"},
		{"Ctrl+F / f", "Open Folder Workspace Browser"},
		{"p", "Open Pipeline Configuration"},
		{"m", "Open Model Lifecycle Manager"},
		{"b", "Run CPU Benchmark Suite vs Docling"},
		{"s", "Manage REST API & Web Studio Server"},
		{"e", "Export active document to target format"},
		{"3", "Inspect active DocumentIR structure tree"},
		{"Ctrl+P or >", "Open Command Palette quick search modal"},
		{"Esc", "Return to Home Dashboard from any screen"},
		{"q or 0", "Quit scanDOC TUI"},
	}

	for _, sc := range shortcuts {
		b.WriteString(styles.ActiveItemStyle.Render(sc[0]) + strings.Repeat(" ", 20-len(sc[0])) + styles.NormalItemStyle.Render(sc[1]) + "\n")
	}

	b.WriteString("\n" + styles.FooterStyle.Render("Press Enter or Esc to return to Home Dashboard."))
	return styles.PanelStyle.Render(b.String())
}

File: src/scandoc/tui/screens/help.py
"""
Help Screen renderer for scanDOC TUI.
"""

from rich.panel import Panel
from rich.table import Table

from scandoc.tui.state import TuiState


def render_help_screen(state: TuiState) -> Panel:
    """Render Keyboard Shortcuts & Help reference screen."""
    table = Table(show_header=True, box=None, padding=(0, 2))
    table.add_column("Shortcut", style="bold cyan", width=14)
    table.add_column("Action Description", style="bold white")

    shortcuts = [
        ("q / Esc", "Quit or Return to Previous Screen"),
        ("?", "Open Help & Shortcut Reference"),
        ("slash", "Focus File Search Filter in Browser"),
        ("o", "Open File Picker Screen"),
        ("f", "Open Folder Browser"),
        ("m", "Open Model Manager"),
        ("p", "Open Pipeline Editor"),
        ("b", "Open Benchmarking Dashboard"),
        ("s", "Open REST Server Manager"),
        ("e", "Open Multi-Format Exporter UI"),
        ("l", "Toggle Live Progress Logs"),
        ("Ctrl+P", "Open Claude-Code-Style Command Palette"),
        ("Space", "Toggle Checkbox or Multi-select Items"),
        ("Enter", "Confirm Selection or Execute Action"),
    ]

    for key, desc in shortcuts:
        table.add_row(f"[{key}]", desc)

    return Panel(
        table,
        title="[bold blue]scanDOC TUI Keyboard Shortcuts & Quick Guide[/bold blue]",
        subtitle="Esc: Return to Previous Screen",
        border_style="cyan",
        padding=(1, 2),
    )
